camel-knative/camel-knative-api/src/main/java/org/apache/camel/component/knative/spi/Knative.java

```diff
@@ -40,13 +37,18 @@
     private Knative() {
     }
 
+    public enum Kind {
```

Comment:
I'm working on the operator to make a version that can work with broker/trigger model. The problem is that the pair `(name, source/sink)` is not sufficient to identify a resource in all cases. There are some edge cases we have to live with, now that `Channel`s have been moved into CRDs. E.g. one many define a `KafkaChannel` named "mychan" and a `InMemoryChannel` named "mychan" as well and we should allow to target both.

Being this an edge case, we can open a issue to solve it later in runtime, but I'm adding support for it in the operator, so I've added not 1 but 3 fields in the `CamelServiceDefinition`:
- `apiVersion`
- `kind`
- this field that you called `kind` but better we call `kind` the Kubernetes Kind. I've called it `consumption`, with possible values `consumer` or `producer`, but it's probably a name worse than `kind`

Just.. let's find a better name for this, I'm fine with the enum values proposed here.

Reply:
I think we can use some "namespace" to distinguish things like:

- `camel.endpoint.kind` for source vs sink
- `knative.kind` and `knative.api-version` for additional kantive metadata

So a service is always identified by is name then some additional filtering logic can be applied on the endpoint side depending on the options provided